### DISCUSSION TITLE
How to implement useSWRPages list with filtration based on prop

### DISCUSSION THREAD

**Discussion by konstaner** [vercel/next.js#13093] [Help]
I implement list of items with pagination using useSWR and useSWRPages. Also I need to refetch list when prop changes.

Here is example: https://codesandbox.io/s/sad-sara-mu5lx?file=/src/App.js

If one or two items loaded and I click on filter buttons result is correct, because count of values is equal. But if I load more items, than it is in filtered list and then click on filter buttons the result is **duplicating**
 
Any ideas how to fix this or may be I should do it in another way?

**Comment by konstaner** [comment]
It seems like there is already fix for this, but not merged yet
https://github.com/zeit/swr/pull/404
https://github.com/zeit/swr/issues/396

**Comment by shuding** [comment]
Hi @konstaner, we have a new API here: https://github.com/vercel/swr/pull/435 and it's released as a beta version `swr@0.3.0-beta.2`. 

Could you please check if it works for you?
